openedx/openedx-platform
# fix: add aria-current attributes to video captions

## Description

This PR fixes the accessibility issues associated with only visually treating the current transcript line. The current implementation for the transcript panel bolds the text that is actively being spoken or skipped to. However, there is no aria attribute present to convey this change to assistive technology. This change impacts learners and course authors.

## Supporting information

JIRA Ticket: [AU-2176 🔒](https://2u-internal.atlassian.net/browse/AU-2176)
> As the video plays, we dynamically change styling of the chunk of the transcript synced with the current position in the timeline of the video playhead by changing it from `<li class>` with `font-weight:400` to `<li class="current">` with `font-weight:700`. This visual information is not currently conveyed to learners using assistive technology.
> 
> We should indicate the status of "current" programmatically so that a learner using a screen reader will have the same information as a sighted user and use that to help with orientation within the transcript. The current visual-only treatment of the current transcript chunk fails to conform with: WCAG SC 1.3.1: Info and Relationships (Level A)
>
> Proposed Solution
>
> We can accomplish this by [programmatically applying the ](https://www.w3.org/TR/wai-aria/#aria-current)aria-current attribute to the current chunk alongside the change to `<li class="current">`.


## Testing instructions

1. Open a course is Studio
2. Add a video
3. Add transcripts to video
4. In the studio preview open the transcript pane
5. Open Dev Tools
6. Click on one of the transcript lines
7. Confirm in Dev Tools that the bolded line has `aria-current` set to `true`
8. Click on another line
9. Confirm in Dev Tools that the previously clicked line has `aria-current` set to `false`
10. Confirm in Dev Tools that the newly clicked line has `aria-current` set to `true`
11. Publish changes
12. Click "View live"
13. Repeat steps 5 through 10 in the student view

## Deadline

None
